crypto/asn1/ameth_lib.c

```diff
@@ -124,24 +125,26 @@ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find_str(ENGINE **pe,
 #endif
         *pe = NULL;
     }
-    for (i = 0; i < EVP_PKEY_asn1_get_count(); i++) {
+    for (i = EVP_PKEY_asn1_get_count(); i-- > 0; ) {
         ameth = EVP_PKEY_asn1_get0(i);
-        if (ameth->pkey_flags & ASN1_PKEY_ALIAS)
-            continue;
         if (((int)strlen(ameth->pem_str) == len)
             && (strncasecmp(ameth->pem_str, str, len) == 0))
-            return ameth;
+            break;
     }
-    return NULL;
+    if (ameth != NULL) {
+        for (;;) {
+            int type = ameth->pkey_base_id;
```

Comment:
side-note, put the type decl up with i and save a line :)  or even just get rid of that local variable since it's only used once.  and put the if test at 138 at 135 revised to be a while loop :)